### ISSUE TITLE
IPython prompt does not return after RecursionError message

### ISSUE THREAD

**Issue by Khris777** [spyder-ide/spyder#5052]
## Description of your problem

After a RecursionError message the IPython prompt does not show up again. 

You can still enter stuff into the console, but it won't do anything, I encounter a similar behaviour as described in issue https://github.com/spyder-ide/spyder/issues/2696 .

By repeatedly doing this it's also possible to completely lock up the console so that it has to be killed.

With the very long RecursionError message I also encounter issue https://github.com/spyder-ide/spyder/issues/2114 .

EDIT ================== 

I am also getting this problem after causing a trivial error after a succesful iteration with a complex object (an object with nested OrderedDictionaries in my real application that I can not post here).

I have not been able to create a working reproducible example though, but there might be a connection to issue https://github.com/spyder-ide/spyder/issues/4221 .

/EDIT ================== 

EDIT 2 ================== 

It seems that this problem is also appearing when no recursion is involved if there are complex and/or huge objects present. Right now there are two large pandas dataframes (11 million lines x 8 columns, and 11 million lines x 26 columns) in my console and any error message fails to return the prompt.

/EDIT 2 ================== 

EDIT 3 ================== 

Upgraded to Spyder 3.2.1, problem persists as described.

/EDIT 3 ================== 

**What steps will reproduce the problem?**

1. Cause a long RecursionError message like this:
```
def iterateThing(x):
    return iterateThing(x+1)
iterateThing(0)
```
2. Try entering stuff manually repeatedly.

3. Execute a line of code using F9.

Try doing these in quick succession and the console might lock up completely.

## Versions and main components

* Spyder Version: 3.2.0
* Python Version: 3.6.2
* Qt Version: 5.8.0
* PyQt Version: 5.8.2
* Operating system: Windows 10

## Dependencies

```
IPython >=4.0    :  6.1.0 (OK)
cython >=0.21    :  0.26 (OK)
jedi >=0.9.0     :  0.10.2 (OK)
nbconvert >=4.0  :  5.2.1 (OK)
numpy >=1.7      :  1.13.1 (OK)
pandas >=0.13.1  :  0.20.3 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pyflakes >=0.6.0 :  1.5.0 (OK)
pygments >=2.0   :  2.2.0 (OK)
pylint >=0.25    :  1.6.5 (OK)
qtconsole >=4.2.0:  4.3.0 (OK)
rope >=0.9.4     :  0.10.5 (OK)
sphinx >=0.6.6   :  1.6.2 (OK)
sympy >=0.7.3    :  None (NOK)
```

**Comment by ccordoba12** [comment]
Thanks for reporting. This is a duplicate of issue #4504 and it's going to be fixed in our 3.2.2 version.

**Comment by ccordoba12** [comment]
In the meantime, you can press `Ctrl+L` to get a prompt back.
